gnomad/assessment/sanity_checks.py

```diff
@@ -190,3 +212,744 @@ def compare_row_counts(ht1: hl.Table, ht2: hl.Table) -> bool:
+    :return: None
+    """
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+
+    filters = t.aggregate(hl.agg.counter(t.filters))
+    logger.info("hl.agg.counter filters: %s", filters)
+
+    filtered_expr = hl.len(t.filters) > 0
+
+    problematic_region_exprs = []
+    if check_lcr:
+        problematic_region_exprs.append(t.info.lcr)
+    if check_segdup:
+        problematic_region_exprs.append(t.info.segdup)
+    if check_nonpar:
+        problematic_region_exprs.append(t.info.nonpar)
+    if check_decoy:
+        problematic_region_exprs.append(t.info.nonpar)
+
+    problematic_region_expr = hl.any(lambda x: x, problematic_region_exprs)
+
+    t = t.annotate(
+        is_filtered=filtered_expr, in_problematic_region=problematic_region_expr
+    )
+
+    def _filter_agg_order(
+        t: Union[hl.MatrixTable, hl.Table],
+        group_exprs: Dict[str, hl.expr.Expression],
+        n_rows: int = None,
+        n_cols: int = None,
+        extra_filter_checks: Optional[Dict[str, hl.expr.Expression]] = None,
+        variant_filter_field: str = "RF",
+    ) -> None:
+        """
+        Perform sanity checks to measure percentages of variants filtered under different conditions.
+
+        :param t: Input MatrixTable or Table.
+        :param group_exprs: Dictionary of expressions to group the Table by.
+        :param n_rows: Number of rows to show.
+        :param n_cols: Number of columns to show.
+        :param extra_filter_checks: Optional dictionary containing filter condition name (key) extra filter expressions (value) to be examined.
+        :param variant_filter_field: String of variant filtration used in the filters annotation on `ht` (e.g. RF, VQSR, AS_VQSR).
+        :return: None
+        """
+        t = t.rows() if isinstance(t, hl.MatrixTable) else t
+        # NOTE: make_filters_sanity_check_expr returns a dict with %ages of variants filtered
+        t.group_by(**group_exprs).aggregate(
+            **make_filters_sanity_check_expr(
+                t, extra_filter_checks, variant_filter_field
+            )
+        ).order_by(hl.desc("n")).show(n_rows, n_cols)
+
+    logger.info(
+        "Checking distributions of filtered variants amongst variant filters..."
+    )
+
+    _filter_agg_order(
+        t, {"is_filtered": t.is_filtered}, variant_filter_field=variant_filter_field,
+    )
+
+    logger.info("Checking distributions of variant type amongst variant filters...")
+    _filter_agg_order(
+        t,
+        {"allele_type": t.info.allele_type},
+        variant_filter_field=variant_filter_field,
+    )
+
+    logger.info(
+        "Checking distributions of variant type and region type amongst variant filters..."
+    )
+    _filter_agg_order(
+        t,
+        {
+            "allele_type": t.info.allele_type,
+            "in_problematic_region": t.in_problematic_region,
+        },
+        50,
+        140,
```

Comment:
These are parameters in ` _filter_agg_order` do you think they should be parameters to `filters_sanity_check` also? If not then do they need to be parameters in ` _filter_agg_order`?

Reply:
I wondered about this too during my review. these two numbers are only used for the slightly more complex shows (calling `_filter_agg_order` with two or more `group_exprs`) and are slightly arbitrary (I took these directly from the v2 code). I guess we could make these parameters to `filters_sanity_check`? maybe something like `large_n_rows: int = 50` and `large_n_cols: int = 140`, with this in the docstring
```
:param large_n_rows: Number of rows to show when showing percentages of filtered variants grouped by multiple conditions. Default is 50.
:param large_n_cols: Number of columns to show when showing percentages of filtered variants grouped by multiple conditions. Default is 140.
```